file.go

```diff
@@ -188,8 +221,23 @@ func (f *File) Update(ctx Context) error {
 				return err
 			}
 		}
+
+		return f.Create(ctx)
 	}
-	return f.Create(ctx)
+
+	if !f.KeepExistingContent {
+		err := f.writeContent(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	err := f.ensureMode(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
nit
```suggestion
	return f.ensureMode(ctx)
```

Reply:
I think I prefer to keep it as a different action in case we add actions later or reorder them.

Reply:
Ok, it makes sense. Let's keep it as it is now  :+1: